guides/source/autoloading_and_reloading_constants.md

```diff
@@ -151,18 +151,17 @@ as you can see, the class object stored in the `User` constant is different afte
 
 It is very important to understand that Ruby does not have a way to truly reload classes and modules in memory, and have that reflected everywhere they are already used. Technically, "unloading" the `User` class means removing the `User` constant via `Object.send(:remove_const, "User")`.
 
-Therefore, if you store a reloadable class or module object in a place that is not reloaded, that value is going to become stale.
+Therefore, if you store a reloadable class or module object in a place that is not reloaded, that value will become stale.
 
-For example, if an initializer stores and caches a certain class object
+For example, if an initializer stores and caches a class object:
 
 ```ruby
 # config/initializers/configure_payment_gateway.rb
 # DO NOT DO THIS.
 $PAYMENT_GATEWAY = Rails.env.production? ? RealGateway : MockedGateway
-# DO NOT DO THIS.
 ```
 
-and `MockedGateway` gets reloaded, `$PAYMENT_GATEWAY` still stores the class object `MockedGateway` evaluated to when the initializer ran. Reloading does not change the class object stored in `$PAYMENT_GATEWAY`.
+`MockedGateway` gets reloaded, `$PAYMENT_GATEWAY` will store the class object `MockedGateway`, which was evaluated when the initializer ran. Reloading does not change the class object stored in `$PAYMENT_GATEWAY`.
```

Comment:
I think this one is confusing. It is important to underline that the content of `$PAYMENT_GATEWAY` does not change ("still stores"), and to refer to the class object the constant holded in the past, when the assignment was performed.

Reply:
I mean "the class object MockedGateway" is ambiguous, because in the example the constant has been reloaded. We are specifically referring to the class object stored when the constant was evaluated (a constant is an expression).

Reply:
Was trying to reword this a few times, WDYT of this?

```
`$PAYMENT_GATEWAY` will store the class object `MockedGateway`, which was evaluated when the initializer ran. So the content stored in ObjectSpace will reference the *initial* class object. When reloading does happen, `$PAYMENT_GATEWAY` will not change.
```

Reply:
Just to understand this properly:

In development `MockedGateway` will get reloaded but when we assign it to `$PAYMENT_GATEWAY` this object will always store the original content, regardless of autoloading.

In production the `RealGateway` is never autoloaded.

Either way using this approach `$PAYMENT_GATEWAY` is always the same, therefore if you want to autoload use the reloader callback?